infrastructure/src/main/java/org/corfudb/infrastructure/CorfuServer.java

```diff
@@ -257,116 +336,167 @@ public static void main(String[] args) throws Exception {
+                        .put(SequencerServer.class, new SequencerServer(serverContext))
+                        .put(LayoutServer.class, new LayoutServer(serverContext))
+                        .put(LogUnitServer.class, new LogUnitServer(serverContext))
+                        .put(ManagementServer.class, new ManagementServer(serverContext))
+                        .build()
+        );
     }
 
     /**
-     * Setup logback logger
-     *  - pick the correct logging level before outputting error messages
-     *  - add serverEndpoint information
+     * Corfu Server initialization.
      *
-     * @param opts command line parameters
-     * @throws JoranException logback exception
+     * @param serverContext Initialized Server Context.
+     * @param serverMap     Server Map with all components.
      */
-    private static void configureLogger(Map<String, Object> opts) {
-        final Logger root = (Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME);
-        final Level level = Level.toLevel(((String) opts.get("--log-level")).toUpperCase());
-        root.setLevel(level);
+    public CorfuServer(@Nonnull ServerContext serverContext,
+                       @Nonnull Map<Class, AbstractServer> serverMap) {
+        this.serverContext = serverContext;
+        this.serverContext.setBindToAllInterfaces(bindToAllInterfaces);
+        this.serverMap = serverMap;
+        router = new NettyServerRouter(new ArrayList<>(serverMap.values()));
+        this.serverContext.setServerRouter(router);
     }
 
     /**
-     * Creates all the components of the Corfu Server.
-     * Starts the Server Router.
+     * Start the Corfu Server by listening on the specified port.
      *
-     * @param opts Options passed by the user.
+     * @return Listening Channel Future.
      */
-    public static void startServer(Map<String, Object> opts, boolean bindToAllInterfaces) {
+    public ChannelFuture start() {
+        bindFuture = startAndListen(serverContext.getBossGroup(),
+                serverContext.getWorkerGroup(),
+                CorfuServer::configureBootstrapOptions,
+                serverContext,
+                router,
+                (String) serverContext.getServerConfig().get("--address"),
+                Integer.parseInt((String) serverContext.getServerConfig().get("<port>")));
+
+        return bindFuture.syncUninterruptibly();
+    }
 
-        int port = Integer.parseInt((String) opts.get("<port>"));
+    @Override
+    public synchronized void close() {
+        if (!close) {
+            log.info("close: Shutting down Corfu server and cleaning resources");
+            close = true;
+            serverContext.close();
+            bindFuture.channel().close().syncUninterruptibly();
+
+            // A executor service to create the shutdown threads
+            // plus name the threads correctly.
+            final ExecutorService shutdownService = Executors.newFixedThreadPool(serverMap.size(),
```

Comment:
This executor seems very suspicious, why not to use at least common thread pool? Best case could be - get rid of this executor